### DISCUSSION TITLE
Threading support

### DISCUSSION THREAD

**Discussion by omniraptorr** [bluesky-social/atproto#2321] [Bluesky Lexicons]
When someone is writing a long thread, they sometimes add e.g. 'x/73' to each post, and i was wondering if that info could be included in the lexicon? 

Basically each post record that's part of a thread would have a field indicating as much, ~~and would include a pointer to the next post in the thread~~.  This would also let the author go on small tangents while still specifying a 'main' thread. In turn, clients would be able to prefetch and render the thread more conveniently for readers.

There would also need to be a ui change in the app to let people write these long explicit threads ofc.

**Comment by DavidBuchanan314** [comment]
Starting from the end of a thread, there's already a complete chain of references back to the root. For example, here's a thread unroller app I made https://retr0.id/stuff/bskythreads/ (just a prototype really).

It's conceptually difficult for a post to reference other posts that might come after it. But I like the idea of an explicitly marked thread.

My suggestion would be, rather than having extra data at the start of the thread, have the extra data at the end of the thread - a boolean flag field that says "this is the end of a thread" - everything else can be worked out from there.

Another option is an entirely separate record, a "thread manifest", that's basically just a list of all the records in the thread. This would enable the prefetching behaviour you describe. In the UI it could function a lot like a repost, e.g:

![image](https://github.com/bluesky-social/atproto/assets/13520633/0a984c46-75d6-4314-8974-3c1f0984ed29)

**Reply by snarfed** [reply]
Agreed! Also, the heuristic of "take the first reply by the author, then take the first reply to _that_ reply by the author, and so on until there aren't any more" will get you the complete thread 99% of the time, even if the author has other replies branching off of the main thread.

Explicit thread markers and manifests might still be useful, but given that heuristic and the existing `replyRef`s, we can probably get a long way without any explicit thread metadata.

**Reply by omniraptorr** [reply]
> It's conceptually difficult for a post to reference other posts that might come after it. But I like the idea of an explicitly marked thread.

my b, hadn't considered that you can't reference future posts. but marking individual posts as part of a thread (without a reference) would alert clients to prioritize displaying the entire thread (and in a social sense maybe also help mitigate context collapse a bit).
